core/src/main/java/org/hisp/dhis/android/core/fileresource/internal/FileResourceUtil.kt

```diff
@@ -71,12 +70,12 @@ internal object FileResourceUtil {
         return writeInputStream(inputStream, destinationFile, sourceFile.length())
     }
 
-    fun saveFileFromResponse(body: ResponseBody, fileResource: FileResource, context: Context): File {
+    fun saveFileFromResponse(body: ByteArray, fileResource: FileResource, context: Context): File {
         val destinationFile = File(
             FileResourceDirectoryHelper.getFileResourceDirectory(context),
             generateFileName(fileResource.uid()!!, getExtension(fileResource.name()!!)),
         )
-        writeInputStream(body.byteStream(), destinationFile, body.contentLength())
```

Comment:
did you check if the method `size.toLong()` returns the same value as `.contentLength()`?

Reply:
Yes, tested on develop and with this branch and the results are the same